lib/index.js

```diff
@@ -21,14 +21,39 @@ function farm (options, path, methods) {
   return api
 }
 
-
-function end (api, callback) {
+function get (api) {
   for (let i = 0; i < farms.length; i++)
     if (farms[i] && farms[i].api === api)
-      return farms[i].farm.end(callback)
-  process.nextTick(callback.bind(null, 'Worker farm not found!'))
+      return farms[i].farm;
+  return null
+}
+
+
+function stdout (api) {
+  let farm = get(api);
+  if (farm)
+    return farm.stdout
+  throw new ReferenceError('Worker farm not found!')
+}
+
+
+function stderr (api) {
+  let farm = get(api);
+  if (farm)
+    return farm.stderr
+  throw new ReferenceError('Worker farm not found!')
+}
+
+
+function end (api, callback) {
+  let farm = get(api);
+  if (farm)
+    return farm.end(callback)
+  process.nextTick(callback.bind(null, new ReferenceError('Worker farm not found!')))
```

Comment:
I added another small change here: to make it more compatible with potentially `Promise`-ified versions, I changed this from a `string` to a `ReferenceError` instance.

Reply:
great catch